src/Evolution/Systems/ScalarWave/BoundaryConditions/ConstraintPreservingSphericalRadiation.hpp

```diff
@@ -0,0 +1,264 @@
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace domain::Tags {
+template <size_t Dim, typename Frame>
+struct Coordinates;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace ScalarWave::BoundaryConditions {
+namespace detail {
+/// The type of spherical radiation boundary condition to impose
+enum class ConstraintPreservingSphericalRadiationType {
+  /// Impose \f$(\partial_t + \partial_r)\Psi=0\f$
+  Sommerfeld,
+  /// Impose \f$(\partial_t + \partial_r + r^{-1})\Psi=0\f$
+  FirstOrderBaylissTurkel,
+  /// Imposes a second-order Bayliss-Turkel boundary condition
+  SecondOrderBaylissTurkel
+};
+
+ConstraintPreservingSphericalRadiationType
+convert_constraint_preserving_spherical_radiation_type_from_yaml(
+    const Options::Option& options);
+}  // namespace detail
+
+/*!
+ * \brief Constraint-preserving spherical radiation boundary condition that
+ * seeks to avoid ingoing constraint violations and radiation.
+ *
+ * The constraint preserving part of the boundary condition imposes the
+ * following condition on the time derivatives of the characteristic fields:
+ *
+ * \f{align*}{
+ *   d_tw^\Psi&\to d_tw^{\Psi}+\lambda_{\Psi}n^i\mathcal{C}_i, \\
+ *   d_tw^0_i&\to d_tw^{0}_i+\lambda_{0}n^jP^k_i\mathcal{C}_{ik},
+ * \f}
+ *
+ * where
+ *
+ * \f{align*}{
+ * P^k{}_i=\delta^k_i-n^kn_i
+ * \f}
+ *
+ * is the projection operator, and \f$d_t w\f$ is the evolved to characteristic
+ * field transformation applied to the time derivatives of the evolved fields.
+ * That is,
+ *
+ * \f{align*}{
+ * d_t w^\Psi&=\partial_t \Psi, \\
+ * d_t w_i^0&=(\delta^k_i-n^k n_i)\partial_t \Phi_k, \\
+ * d_t w^{\pm}&=\partial_t\Pi\pm n^k\partial_t\Phi_k - \gamma_2\partial_t\Psi.
+ * \f}
+ *
+ * The constraints are defined as:
+ *
+ * \f{align*}{
+ *  \mathcal{C}_i&=\partial_i\Psi - \Phi_i=0, \\
+ *  \mathcal{C}_{ij}&=\partial_{[i}\Phi_{j]}=0
+ * \f}
+ *
+ * Radiation boundary conditions impose a condition on \f$\Pi\f$ or its time
+ * derivative. We denote the boundary condition value of the time derivative of
+ * \f$\Pi\f$ by \f$\partial_t\Pi^{\mathrm{BC}}\f$. With this, we can impose
+ * boundary conditions on the time derivatives of the evolved variables follows:
+ *
+ * \f{align*}{
+ * \partial_{t} \Psi&\to\partial_{t}\Psi +
+ *                    \lambda_\Psi n^i \mathcal{C}_i, \\
+ * \partial_{t}\Pi&\to\partial_{t}\Pi-\left(\partial_t\Pi
+ *                  - \partial_t\Pi^{\mathrm{BC}}\right)
+ *                  +\gamma_2\lambda_\Psi n^i \mathcal{C}_i
+ *                  =\partial_t\Pi^{\mathrm{BC}}
+ *                  +\gamma_2\lambda_\Psi n^i \mathcal{C}_i, \\
+ * \partial_{t}\Phi_i&\to\partial_{t}\Phi_i+
+ *                     \lambda_0n^jP^k{}_i\mathcal{C}_{jk}
+ *   = \partial_{t}\Phi_i+ \lambda_0n^j \mathcal{C}_{ji}.
+ * \f}
+ *
+ * Below we assume the normal vector \f$n^i\f$ is the radial unit normal vector.
+ * That is, we assume the outer boundary is spherical. A Sommerfeld radiation
+ * condition is given by
```

Comment:
[optional, if you think it would be useful] consider adding refs for Sommerfeld, Bayliss-Turkel statements here and below.